### PR TITLE
Backend CI cleanups and improvements

### PR DESCRIPTION
**Pull request by dapolach**
- no GITHUB reference from gradle file
- docker image build just once

### CODE REVIEW

File: .github/workflows/backend_publish.yml [hunk 1]
```diff
@@ -34,13 +35,11 @@ jobs:
 
     - name: Build and Publish Docker Image
       working-directory: backend
-      env:
```
Comment (otahirs) [inline .github/workflows/backend_publish.yml:37]: Tady bych nastavil             DOCKER_REGISTRY_USERNAME
a pod

**Review by otahirs — CHANGES_REQUESTED** [review]

File: .github/workflows/backend_publish.yml [hunk 2]
```diff
@@ -34,13 +35,11 @@ jobs:
...
-        GITHUB_ACTOR: ${{ github.actor }}
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
         # Build the Docker image and tag it with both the 'latest', commit hash, and tag if exist
         ./gradlew bootBuildImage --imageName=ghcr.io/zabiny/klabis-be:latest
-        ./gradlew bootBuildImage --imageName=ghcr.io/zabiny/klabis-be:$(git rev-parse --short HEAD)
+        docker tag ghcr.io/zabiny/klabis-be:latest ghcr.io/zabiny/klabis-be:$(git rev-parse --short HEAD) 
+        docker push -a ghcr.io/zabiny/klabis-be
```
Comment (otahirs) [inline .github/workflows/backend_publish.yml:42]: docker commands will not work without logging in first 
https://github.com/docker/login-action

Comment (dapolach) [reply .github/workflows/backend_publish.yml:42]: `docker` is [logged in|https://github.com/zabiny/klabis/pull/248/files/6e2fac83ca4778407ec2d99ff1a1642070d0019d#diff-1ada79d11a8d53da3cf6fafbab70a6a7d981c056dab14e2baa0b34b06bf11aeaL28] ... you wrote it ;) 

pipeline works - https://github.com/zabiny/klabis/actions/runs/10862180442/job/30144910105

File: .github/workflows/backend_publish.yml (hunk 1 above)
Comment (dapolach) [reply .github/workflows/backend_publish.yml:37]: Proc? Neni pro to zadny duvod.

**Comment by dapolach** [comment]
Pipeline jak je tady nastavena tak funguje - je otestovana napr. tady: https://github.com/zabiny/klabis/actions/runs/10862180442/job/30144910105

File: .github/workflows/backend_publish.yml (hunk 2 above)
Comment (dapolach) [reply .github/workflows/backend_publish.yml:42]: Pipeline jak je tady nastavena tak funguje - je otestovana napr. tady: https://github.com/zabiny/klabis/actions/runs/10862180442/job/30144910105  

Ostatne vypis vyse je prave z toho testu...

Comment (otahirs) [reply .github/workflows/backend_publish.yml:42]: > `docker` is logged in from previous step in pipeline ... you wrote that pipeline... ;)
> 
> image

I got bamboozled by myself. It was not needed in my pipeline as the publish was done by the gradle task with credentials, so consequently I did not see it the _diff_ so I thought there is missing authorization. :clown_face:

File: .github/workflows/backend_publish.yml (hunk 1 above)
Comment (otahirs) [reply .github/workflows/backend_publish.yml:37]: you are right, i missed that that the gradle task is now used for build only

**Comment by otahirs** [comment]
sorry for my confusion, checking PR from mobile phone might not be one of the best of my ideas

**Pull request merged**